src/wp-includes/class-wp-editor.php

```diff
@@ -1556,9 +1556,7 @@ public static function print_tinymce_scripts() {
 			script_concat_settings();
 		}
 
-		wp_print_scripts( array( 'wp-tinymce' ) );
-
-		echo "<script type='text/javascript'>\n" . self::wp_mce_translation() . "</script>\n";
+		do_action( 'print_tinymce_scripts' );
```

Comment:
I wasn't sure how to add tests for this change, would love some pointers on that.

I tried testing around `did_action( 'print_tinymce_scriots' )` but it seems like each test isn't running with a brand new WP so `did_action` returns `true` before doing anything inside the test body. Is there a way to observe whether this action is performed when I call `_WP_Editors::print_tinymce_scripts()` directly? Is this a worthwhile test?

Reply:
Hm, @saramarcondes, I can't reproduce this `did_action` behavior in tests. It clears up for me. Here is my attempt to reduce the issue: 

```php
<?php

class Test_Did_Action extends WP_UnitTestCase {
	public function test_did_action_should_be_two_if_action_fired_twice() {
		do_action( 'wink' );
		do_action( 'wink' );
		$this->assertEquals( 2, did_action( 'wink' ) );
	}

	public function test_did_action_should_be_one_if_action_fired_once() {
		do_action( 'wink' );
		$this->assertEquals( 1, did_action( 'wink' ) );
	}

	public function test_did_action_should_be_zero_if_no_action_fired() {
		$this->assertEquals( 0, did_action( 'wink' ) );
	}
}
```

If WordPress doesn't clean its globals (this is where `do_action`/`did_action` [keep information](https://github.com/WordPress/wordpress-develop/blob/3659b767f1f8a4d3ca08f6d8310963b9d3a73b08/src/wp-includes/plugin.php#L493-L501)) at least one of the cases would have failed (if we assume they could have been executed in any order).

If something else is firing the action (code outside of our control), maybe we can test whether the difference of the return value of `did_action` before and after our code is what we want? Zero if we don't expect our code to fire the action, or one if we expect it to fire once?

Reply:
Ah! I believe the issue is that I incorrectly assumed `did_action` returns `boolean` and `assertFalse( 0 )` isn't going to pass 😀

Reply:
Haha, I can't blame you – I wish anything starting with `did_` only returned a boolean, but 🤷‍♂️